### ISSUE TITLE
super within function closure does not bind correctly

### ISSUE THREAD

**Issue by eventualbuddha** [6to5/6to5#425]
```js
class Req {
  post(request) {
    return handleErrors(function() {
      return super.post(request);
    });
  }
}
```

In this case, `super.post(request)` should call with its `this` binding set to whatever it is inside `post` (likely an instance of `Req`), but instead it's whatever `this` is inside the function closure passed to `handleErrors`.

**Comment by ms88privat** [comment]
+1 that is a big issue. Thank you

**Comment by sebmck** [comment]
Fixed as of 2.11.0
